core/server/apps/amp/lib/helpers/amp_content.js

```diff
@@ -67,9 +68,28 @@ function ampContent() {
         };
 
     return Promise.props(amperizeHTML).then(function (result) {
+        var $;
+
+        // our Amperized HTML
         ampHTML = result.amperize || '';
 
-        // let's sanitize our HTML!!!
+        // Use cheerio to traverse through HTML and make little clean-ups
+        $ = cheerio.load(ampHTML);
+
+        // We have to remove source children in video, as source
+        // is whitelisted for audio, but causes validation
+        // errors in video, because video will be stripped out.
+        // @TODO: remove this, when Amperize support video transform
+        $('video').children('source').remove();
```

Comment:
The `video` tags gets stripped out later. The problem here is, that the `source` tags will stay. But you can have them for `audio` as well:

```
<audio controls="controls">
  Your browser does not support the <code>audio</code> element.
  <source src="http://developer.mozilla.org/@api/deki/files/2926/=AudioTest.ogg" type="audio/wav">
</audio>
```
^ In this case, `audio` will be transformed to `amp-audio` and therefore, `source` tags are valid.

But see `video`:
```
<video width="480" controls poster="https://archive.org/download/WebmVp8Vorbis/webmvp8.gif" >
  <source src="https://archive.org/download/WebmVp8Vorbis/webmvp8.webm" type="video/webm">
  <source src="https://archive.org/download/WebmVp8Vorbis/webmvp8_512kb.mp4" type="video/mp4">
  <source src="https://archive.org/download/WebmVp8Vorbis/webmvp8.ogv" type="video/ogg">
  Your browser doesn't support HTML5 video tag.
</video>
```
^ Here, `video` will be stripped out later on by `sanitize` and unfortunately, I didn't find a way in `sanitize` to check for children elements. Because, when `video` is removed, the `source` tags remain, and they will cause validation errors.